packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -215,8 +215,35 @@ class TextMapPropagator {
   }
 
   _extractSpanContext (carrier) {
+    let spanContext = null
     for (const extractor of this._config.tracePropagationStyle.extract) {
-      let spanContext = null
+      if (spanContext !== null) {
+        if (this._config.tracePropagationExtractFirst) {
+          return spanContext
+        }
+
+        if (extractor !== 'tracecontext') {
+          continue
+        }
+
+        const w3cCtx = this._extractTraceparentContext(carrier)
+
+        if (w3cCtx !== null && spanContext.toTraceId(true) === w3cCtx.toTraceId(true) &&
```

Comment:
Should we add a comment to document this behavior or encapsulate it in a method.